### PR TITLE
jackbox-utility: init at 1.3.7+1

### PR DESCRIPTION
**Pull request by jpeterburs**
## Description of changes

JackboxUtility is an app to download patches and launch Jackbox games.
With this app, one is able to download translation patches for the Jackbox games.

GitHub: https://github.com/jackboxutility/jackboxutility
Homepage: https://jackboxutility.com/

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/ja/jackbox-utility/package.nix [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+      Exec=$out/bin/${pname}
+      Icon=$out/bin/data/flutter_assets/assets/logo.png
+      Terminal=false
+      Comment=An app to download patches and launch Jackbox games
+    EOF
+    chmod +x $out/share/applications/${pname}.desktop
+  '';
+
+  meta = with lib; {
+    description = "An app to download patches and launch Jackbox games ";
```
Comment (Aleksanaa) [inline pkgs/by-name/ja/jackbox-utility/package.nix:46]:
```suggestion
    description = "An app to download patches and launch Jackbox games";
```

File: pkgs/by-name/ja/jackbox-utility/package.nix [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    homepage = "https://github.com/JackboxUtility/JackboxUtility";
```
Comment (Aleksanaa) [inline pkgs/by-name/ja/jackbox-utility/package.nix:47]:
```suggestion
    homepage = "https://jackboxutility.com";
```

File: pkgs/by-name/ja/jackbox-utility/package.nix [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    license = licenses.gpl3;
```
Comment (Aleksanaa) [inline pkgs/by-name/ja/jackbox-utility/package.nix:48]:
```suggestion
    license = licenses.gpl3Plus;
```

`gpl3` is deprecated.

File: pkgs/by-name/ja/jackbox-utility/package.nix [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    maintainers = with maintainers; [ jpeterburs ];
+    mainProgram = "jackbox-utility";
```
Comment (Aleksanaa) [inline pkgs/by-name/ja/jackbox-utility/package.nix:50]:
```suggestion
    mainProgram = "jackbox-utility";
    platforms = [ "x86_64-linux" ];
    sourceProvenance = with sourceTypes; [ binaryNativeCode ];
```

File: pkgs/by-name/ja/jackbox-utility/package.nix [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+    mkdir -p $out/bin $out/share/applications
+    unzip -d $out/bin $src
+    mv $out/bin/JackboxUtility $out/bin/${pname}
+    chmod +x $out/bin/${pname}
+
+    runHook postInstall
+  '';
+
+  postInstall = ''
+    cat << EOF > $out/share/applications/${pname}.desktop
```
Comment (Aleksanaa) [inline pkgs/by-name/ja/jackbox-utility/package.nix:32]: You can do this:

```nix
  desktopItems = [
    (makeDesktopItem {
      name = "JackboxUtility";
      exec = "jackbox-utility";
      comment = "An app to download patches and launch Jackbox games";
      desktopName = "JackboxUtility";
      categories = [ "Game" ];
    })
  ];
```

And add `copyDesktopItems` to `nativeBuildInuts`

File: pkgs/by-name/ja/jackbox-utility/package.nix [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+      [Desktop Entry]
+      Version=${version}
+      Type=Application
+      Name=JackboxUtility
+      Exec=$out/bin/${pname}
+      Icon=$out/bin/data/flutter_assets/assets/logo.png
```
Comment (Aleksanaa) [inline pkgs/by-name/ja/jackbox-utility/package.nix:38]: It's better to link icon to a standard path, e.g. `$out/share/pixmaps/jackbox-utility.png` and write `icon = jackbox-utility`

File: pkgs/by-name/ja/jackbox-utility/package.nix [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+  dontUnpack = true;
+
+  nativeBuildInputs = [ unzip ];
+
+  installPhase = ''
+    runHook preInstall
+
+    mkdir -p $out/bin $out/share/applications
+    unzip -d $out/bin $src
+    mv $out/bin/JackboxUtility $out/bin/${pname}
```
Comment (Aleksanaa) [inline pkgs/by-name/ja/jackbox-utility/package.nix:25]: Please do not reference `pname` unnecessarily.

**Review by Aleksanaa — APPROVED** [review]
Looks good, but would be better to build from source

File: pkgs/by-name/ja/jackbox-utility/package.nix (hunk 7 above)
Comment (jpeterburs) [reply pkgs/by-name/ja/jackbox-utility/package.nix:25]: Thought it would be better to reference than to repeat it, but makes sense

**Comment by jpeterburs** [comment]
I'm new to this, and I wanted to use this software in my NixOS installation in the future, and don't really know that much about Dart and/or flutter.

**Comment by jpeterburs** [comment]
Version `1.3.7+2` has been release.

Full Changelog: https://github.com/JackboxUtility/JackboxUtility/releases/tag/1.3.7%2B2

**Pull request closed without merging**